src/sidebar/components/test/PendingUpdatesNotification-test.js

```diff
@@ -0,0 +1,173 @@
+import { mount } from 'enzyme';
+import sinon from 'sinon';
+
+import { promiseWithResolvers } from '../../../shared/promise-with-resolvers';
+import PendingUpdatesNotification, {
+  $imports,
+} from '../PendingUpdatesNotification';
+
+describe('PendingUpdatesNotification', () => {
+  let fakeSetTimeout;
+  let fakeClearTimeout;
+  let fakeStreamer;
+  let fakeStore;
+
+  beforeEach(() => {
+    fakeSetTimeout = sinon.stub();
+    fakeClearTimeout = sinon.stub();
+    fakeStreamer = {
+      applyPendingUpdates: sinon.stub(),
+    };
+    fakeStore = {
+      pendingUpdateCount: sinon.stub().returns(3),
+      hasPendingUpdates: sinon.stub().returns(true),
+    };
+
+    $imports.$mock({
+      '../store': {
+        useSidebarStore: () => fakeStore,
+      },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  function createComponent(container) {
+    return mount(
+      <PendingUpdatesNotification
+        streamer={fakeStreamer}
+        setTimeout_={fakeSetTimeout}
+        clearTimeout_={fakeClearTimeout}
+      />,
+      { attachTo: container },
+    );
+  }
+
+  function notificationIsCollapsed(wrapper) {
+    return wrapper.exists('[data-testid="collapsed-notification"]');
+  }
+
+  /**
+   * To avoid delaying tests too much, this stubs fakeSetTimeout behavior so
+   * that it schedules a real 1ms timeout, and resolves a promise when finished.
+   * That keeps the async nature of the timeout without affecting test execution
+   * times.
+   */
+  function timeoutAsPromise() {
+    const { resolve, promise } = promiseWithResolvers();
+    fakeSetTimeout.callsFake(callback =>
+      setTimeout(() => {
+        callback();
+        resolve();
+      }, 1),
+    );
+
+    return promise;
+  }
+
+  it('does not render anything while there are no pending updates', () => {
+    fakeStore.hasPendingUpdates.returns(false);
+    const wrapper = createComponent();
+
+    assert.isEmpty(wrapper);
+  });
+
+  it('loads full notification first, and collapses it after a timeout', async () => {
+    const promise = timeoutAsPromise();
+    const wrapper = createComponent();
+
+    // Initially, it shows full notification
+    assert.isTrue(wrapper.exists('[data-testid="full-notification"]'));
+    assert.isFalse(notificationIsCollapsed(wrapper));
+    assert.calledOnce(fakeSetTimeout);
+
+    await promise; // Wait for timeout callback to be invoked
+    wrapper.update();
```

Comment:
`wrapper.update`'s main function is to sync the wrapper with the rendered DOM, which is not immediately updated after the timeout fires, however it does also flush any pending state updates first. Since the state update is queued inside the timeout callback, this is OK.